### ISSUE TITLE
NotFoundError (see above for traceback): Key Bottleneck/BatchNorm/beta not found in checkpoint

### ISSUE THREAD

**Issue by chensteven** [davidsandberg/facenet#850]
I'm trying to generate encodings for the MegaFace and Facescrub datasets for the MegaFace Challenge using the FaceNet (InceptionResNet_v1 model). When I try to restore the 20180402-114759 model, I keep getting an error like this: 

```
2018-08-20 20:03:34.567868: W tensorflow/core/framework/op_kernel.cc:1198] Not found: Key Bottleneck/BatchNorm/beta not found in checkpoint
2018-08-20 20:03:34.571549: W tensorflow/core/framework/op_kernel.cc:1198] Not found: Key Bottleneck/BatchNorm/moving_mean not found in checkpoint
2018-08-20 20:03:34.574023: W tensorflow/core/framework/op_kernel.cc:1198] Not found: Key Bottleneck/BatchNorm/moving_variance not found in checkpoint
2018-08-20 20:03:34.574080: W tensorflow/core/framework/op_kernel.cc:1198] Not found: Key Bottleneck/weights not found in checkpoint
...
Caused by op 'save/RestoreV2', defined at:
  File "facenet_to_bin.py", line 223, in <module>
    main(parse_argument(sys.argv[1:]))
  File "facenet_to_bin.py", line 139, in main
    saver = tf.train.Saver(restore_vars, max_to_keep=3)
  File "/home/schen/.cv/lib/python3.5/site-packages/tensorflow/python/training/saver.py", line 1239, in __init__
    self.build()
  File "/home/schen/.cv/lib/python3.5/site-packages/tensorflow/python/training/saver.py", line 1248, in build
    self._build(self._filename, build_save=True, build_restore=True)
  File "/home/schen/.cv/lib/python3.5/site-packages/tensorflow/python/training/saver.py", line 1284, in _build
    build_save=build_save, build_restore=build_restore)
  File "/home/schen/.cv/lib/python3.5/site-packages/tensorflow/python/training/saver.py", line 765, in _build_internal
    restore_sequentially, reshape)
  File "/home/schen/.cv/lib/python3.5/site-packages/tensorflow/python/training/saver.py", line 428, in _AddRestoreOps
    tensors = self.restore_op(filename_tensor, saveable, preferred_shard)
  File "/home/schen/.cv/lib/python3.5/site-packages/tensorflow/python/training/saver.py", line 268, in restore_op
    [spec.tensor.dtype])[0])
  File "/home/schen/.cv/lib/python3.5/site-packages/tensorflow/python/ops/gen_io_ops.py", line 1031, in restore_v2
    shape_and_slices=shape_and_slices, dtypes=dtypes, name=name)
  File "/home/schen/.cv/lib/python3.5/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/home/schen/.cv/lib/python3.5/site-packages/tensorflow/python/framework/ops.py", line 3160, in create_op
    op_def=op_def)
  File "/home/schen/.cv/lib/python3.5/site-packages/tensorflow/python/framework/ops.py", line 1625, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

NotFoundError (see above for traceback): Key Bottleneck/BatchNorm/beta not found in checkpoint
         [[Node: save/RestoreV2 = RestoreV2[dtypes=[DT_FLOAT], _device="/job:localhost/replica:0/task:0/device:CPU:0"](_arg_save/Const_0_0, save/RestoreV2/tensor_names, save/RestoreV2/shape_and_slices)]]
         [[Node: save/RestoreV2_96/_937 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, tensor_name="edge_1924_save/RestoreV2_96", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]
```

I tried to print all the global and local variables from `tf.get_collection(tf.GraphKeys.GLOBAL_VARIABLES` and was able to see Bottleneck/BatchNorm/beta in the collection.

Do anyone know how to debug this? My first intuition is that it might has to do with loading the model properly using tf.slim.
